src/archive.rs

```diff
@@ -430,26 +431,68 @@ impl<'a> EntriesFields<'a> {
                     ));
                 }
                 pax_extensions = Some(EntryFields::from(entry).read_all()?);
+                // This entry has two headers.
+                // Keep pax_extensions for the next ustar header.
+                processed -= 1;
                 continue;
             }
 
             let mut fields = EntryFields::from(entry);
-            fields.long_pathname = gnu_longname;
-            fields.long_linkname = gnu_longlink;
             fields.pax_extensions = pax_extensions;
+            // False positive: unused assignment
+            // https://github.com/rust-lang/rust/issues/22630
```

Comment:
It looks like this has been fixed so we should be able to drop the assignment.

Reply:
Dropping comments only.

Reply:
We don't have extra assignments here.